src/objects/strategy-cards/strategy-card.js

```diff
@@ -128,6 +49,136 @@ function createStrategyCardUi(card, widgetFactory, height, color) {
     }
 }
 
+/**
+ * Builder class for registering a new strategy card object by adding event
+ * handlers as well as their removal in case of the destruction of the object.
+ */
+class RegisterStrategyCardUI {
+    /**
+     *
+     * @param {object} [data]
+     * @param {GameObject} [data.card] The object instance of the strategy card
+     * @param {function} [data.widgetFactory] A factory creating the UI for each player
+     * @param {Integer} [data.height] Height of the UI for each player
+     * @param {Color} [data.color] Color of the UI for each player
+     * @param {function} [data.onStrategyCardPlayed] Event handler of the "Play" button pressed.
+     * This function only is required if the Play button should do more than just creating a UI.
+     * The whole UI creation should belong into the widgetFactory to separate concerns.
+     * @param {function} [data.onStrategyCardSelectionDone] Event handler for the closing event.
+     */
+    constructor(data) {
+        this._card = data?.card;
+        this._widgetFactory = data?.widgetFactory;
+        this._height = data?.height;
+        this._color = data?.color;
+        this._onStrategyCardPlayed = data?.onStrategyCardPlayed;
+        this._onStrategyCardSelectionDone = data?.onStrategyCardSelectionDone;
+    }
+
+    setCard(value) {
+        this._card = value;
+        return this;
+    }
+
+    setWidgetFactory(value) {
+        this._widgetFactory = value;
+        return this;
+    }
+
+    setHeight(value) {
+        this._height = value;
+        return this;
+    }
+
+    setColor(value) {
+        this._color = value;
+        return this;
+    }
+
+    setOnStrategyCardPlayed(value) {
+        this._onStrategyCardPlayed = value;
+        return this;
+    }
+
+    setOnStrategyCardSelectionDone(value) {
+        this._onStrategyCardSelectionDone = value;
+        return this;
+    }
+
+    /**
+     * Actual registration of the strategy card UI creation.
+     * At this point it is mandatory that <code>object</code>, <code>widgetFactory</code> and
+     * <code>height</code> are set.
+     */
+    register() {
+        // only on registration is allowed at this time
+        assert(!this._registered);
+
+        // the refObject is undefined in a test environment
+        this._card === undefined && world.__isMock
+            ? new GameObject()
+            : this._card;
```

Comment:
This looks off, you mean to be assigning to this._card here?

Reply:
messed this one up. Seems that my current tests are no longer in need of this, but I keep it in since my next test set will require it again.

Reply:
and fixed :D